packages/ketcher-core/src/application/editor/operations/rxn/RxnArrowResize.ts

```diff
@@ -14,69 +14,106 @@
  * limitations under the License.
  ***************************************************************************/
 
-import Base from '../base'
-import { OperationType } from '../OperationType'
-import { Scale } from 'domain/helpers'
+import { ReStruct } from 'application/render'
+import assert from 'assert'
 import { RxnArrow, Vec2 } from 'domain/entities'
+import { Scale } from 'domain/helpers'
 import { tfx } from 'utilities'
+import { OperationType } from '../OperationType'
+import Base from '../base'
+
+export const ARROW_MAX_SNAPPING_ANGLE = Math.PI / 12 // 15°
+
 interface RxnArrowResizeData {
   id: number
-  d: any
+  d: Vec2
   current: Vec2
-  anchor: Vec2
+  anchor: Vec2 | null
   noinvalidate: boolean
 }
 export class RxnArrowResize extends Base {
   data: RxnArrowResizeData
+  isSnappingEnabled: boolean
 
   constructor(
     id: number,
-    d: any,
+    d: Vec2,
     current: Vec2,
-    anchor: any,
-    noinvalidate: boolean
+    anchor: Vec2 | null,
+    noinvalidate: boolean,
+    isSnappingEnabled: boolean
   ) {
     super(OperationType.RXN_ARROW_RESIZE)
     this.data = { id, d, current, anchor, noinvalidate }
+    this.isSnappingEnabled = isSnappingEnabled
   }
 
-  execute(restruct: any): void {
+  execute(restruct: ReStruct): void {
     const struct = restruct.molecule
     const id = this.data.id
-    const d = this.data.d
+    let d = this.data.d
     const current = this.data.current
     const item = struct.rxnArrows.get(id)
+    const reItem = restruct.rxnArrows.get(id)
+    assert(item != null && reItem != null)
     const anchor = this.data.anchor
     if (anchor) {
       const previousPos0 = item.pos[0].get_xy0()
       const previousPos1 = item.pos[1].get_xy0()
       let middlePoint
-      let reItem
 
       if (RxnArrow.isElliptical(item)) {
-        reItem = restruct.rxnArrows.get(id)
         ;[, , middlePoint] = reItem.getReferencePoints()
       }
 
       if (
+        /**
```

Comment:
Proposal:
to draw an image, describe it with words and put it in some kind of README.md file in this folder.
I guess, we can document such a behaviour and complex logic in our application.